crates/opencascade/src/primitives/face.rs

```diff
@@ -213,6 +214,42 @@ impl Face {
         Solid::from_solid(result_solid)
     }
 
+    /// Sweep the face along a path, modulated by a function, to produce a solid
+    #[must_use]
+    pub fn sweep_along_with_radius_values(
+        &self,
+        path: &Wire,
+        radius_values: impl IntoIterator<Item = (f64, f64)>,
+    ) -> Solid {
+        let radius_values: Vec<_> = radius_values.into_iter().collect();
+        let mut array = ffi::TColgp_Array1OfPnt2d_ctor(1, radius_values.len() as i32);
+
+        for (index, (t, radius)) in radius_values.into_iter().enumerate() {
+            array.pin_mut().SetValue(index as i32 + 1, &make_point2d(dvec2(t, radius)));
+        }
+
+        let mut interpol = ffi::Law_Interpol_ctor();
+        let is_periodic = false;
+        interpol.pin_mut().Set(&array, is_periodic);
+        let law_function = ffi::Law_Interpol_into_Law_Function(interpol);
+        let law_handle = ffi::Law_Function_to_handle(law_function);
```

Comment:
I'm usually against premature de-duplication, but this code is complicated enough that we should probably extract parts to some functions.

* One could take a `radius_values: impl IntoIterator<Item = (f64, f64)>,` and return a `UniquePtr<HandleLawFunction>`
* Another that takes a profile `Wire` and returns a `UniquePtr<BRepOffsetAPI_MakePipeShell>`.

The main difference in code paths is one calls `make_solid()` and the other doesn't.

Reply:
Done! :)